### PR TITLE
add fetchInitialRevisions

### PR DESCRIPTION
**Pull request by ghedamat**
this implements https://github.com/ember-cli/ember-cli-deploy/issues/338

### CODE REVIEW

**Comment by ghedamat** [comment]
@lukemelia initially I merged `revisionData` from `fetchInitialRevisions` manually but this https://github.com/ember-cli/ember-cli-deploy/blob/master/lib/models/pipeline.js#L168-L172 should do that for us right?

**Comment by lukemelia** [comment]
@ghedamat While writing 0.6.x docs, I rethought this a little (see results at http://ember-cli.com/ember-cli-deploy/docs/v0.6.x/pipeline-hooks/)

```
fetchInitialRevisions -->  returns an object (or a promise resolving to
                           one) that has an `initialRevisions` property
                           whose value is an array of revision objects.[1]
                           i.e. `{ initialRevisions: [...] }`
```

and

```
            \- fetchRevisions  returns an object (or a promise resolving
             \                 to one) that has a `revisions` property
              \                whose value is an array of revision
               \               objects.[1] i.e. `{ revisions: [...] }`
```

**Comment by lukemelia** [comment]
This would indeed be a breaking change. We need to think about backwards compatibility and/or messaging.

**Comment by ghedamat** [comment]
@lukemelia I tend to agree, if we keep the properties separate we don't need to break  any existing plugins and we can "reserve" the property names in the docs as you did

I'll update the PR and do the other ones later this week

**Comment by ghedamat** [comment]
@lukemelia should be good now and non breaking

File: tests/unit/index-nodetest.js [hunk 1]
```diff
@@ -575,8 +575,8 @@ describe('redis plugin', function() {
     });
   });
 
-  describe('fetchRevisions hook', function() {
-    it('fills the revisions variable on context', function() {
+  describe('fetchInitialRevisions hook', function() {
+    it('fills the revisionsData.initial variable on context', function() {
```
Comment (lukemelia) [inline tests/unit/index-nodetest.js]: test name is no longer correct

Comment (ghedamat) [inline tests/unit/index-nodetest.js]: right, will fix soon

**Pull request merged**